### PR TITLE
Added fix for CVE-2022-41720 for Go 1.17

### LINKED ISSUE #675: os, net/http: avoid escapes from os.DirFS and http.Dir on Windows CVE-2022-41720 - [eks go1.17 backport]
A backport to EKS Go 1.17 is [requested by markapruett](https://github.com/aws/eks-distro-build-tooling/issues/671#issuecomment-1341499171)

### PR DESCRIPTION
**Pull request by kschumy**
*Issue #, if available:*
closes #675 

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: projects/golang/go/1.17/patches/0005-go1.17.13-eks-os-net-http-avoid-escapes-from-os.DirF.patch [hunk 1]
```diff
@@ -0,0 +1,513 @@
+From 21d77ea72e5023d7df5769ee97430ae326009639 Mon Sep 17 00:00:00 2001
+From: Damien Neil <dneil@google.com>
+Date: Thu, 10 Nov 2022 12:16:27 -0800
+Subject: [PATCH 5/9] [go1.17.13-eks] os, net/http: avoid escapes from os.DirFS
+ and http.Dir on Windows
+
```
Comment (danbudris) [inline projects/golang/go/1.17/patches/0005-go1.17.13-eks-os-net-http-avoid-escapes-from-os.DirF.patch:6]: let's add the standard header we've been using for these patch files here as well

Comment (danbudris) [reply projects/golang/go/1.17/patches/0005-go1.17.13-eks-os-net-http-avoid-escapes-from-os.DirF.patch:6]: see https://github.com/aws/eks-distro-build-tooling/tree/main/projects/golang/go for an example

**Comment by rcrozean** [comment]
We also need to update the .spec file in the rpmbuild/spec folder

File: projects/golang/go/1.17/patches/0005-go1.17.13-eks-os-net-http-avoid-escapes-from-os.DirF.patch [hunk 2]
```diff
@@ -0,0 +1,513 @@
...
+Do not permit access to Windows reserved device names (NUL, COM1, etc.)
```
Comment (kschumy) [inline projects/golang/go/1.17/patches/0005-go1.17.13-eks-os-net-http-avoid-escapes-from-os.DirF.patch:7]:
```suggestion
# AWS EKS 
Backported To: go-1.17.13-eks
Backported On: Fri, 16 Dec 2022
Backported By: ksschumy@amazon.com
Backported From: release-branch.go1.18
Upstream Source Commit: https://github.com/golang/go/commit/7013a4f5f816af62033ad63dd06b77c30d7a62a7

# Original Information

Do not permit access to Windows reserved device names (NUL, COM1, etc.)
```

File: projects/golang/go/1.17/rpmbuild/SPECS/golang.spec [hunk 3]
```diff
@@ -157,6 +157,7 @@ Patch1:		0001-go-1.17.13-eks-archive-tar-limit-size-of-head.patch
 Patch2:		0002-go-1.17.13-eks-net-http-httputil-avoid-query-parameter-smuggling.patch
 Patch3:		0003-go-1.17.13-eks-regexp-limit-size-of-parsed-regexps.patch
 Patch4:     0004-go-1.17.13-eks-syscall-os-exec-reject-environ.patch
+Patch5:     0005-go-1.17.13-os-net-http-avoid-escapes-from-os.Dir.patch
```
Comment (rcrozean) [inline projects/golang/go/1.17/rpmbuild/SPECS/golang.spec:160]:
```suggestion
Patch5:     0005-go1.17.13-eks-os-net-http-avoid-escapes-from-os.DirF.patch
```

**Review by rcrozean — CHANGES_REQUESTED** [review]

**Review by rcrozean — APPROVED** [review]

File: projects/golang/go/1.17/rpmbuild/SPECS/golang.spec [hunk 4]
```diff
@@ -157,6 +157,7 @@ Patch1:		0001-go-1.17.13-eks-archive-tar-limit-size-of-head.patch
 Patch2:		0002-go-1.17.13-eks-net-http-httputil-avoid-query-parameter-smuggling.patch
 Patch3:		0003-go-1.17.13-eks-regexp-limit-size-of-parsed-regexps.patch
 Patch4:     0004-go-1.17.13-eks-syscall-os-exec-reject-environ.patch
+Patch5:     0005-go1.17.13-eks-os-net-http-avoid-escapes-from-os.DirF.patch
```
Comment (markapruett) [inline projects/golang/go/1.17/rpmbuild/SPECS/golang.spec:160]:
```suggestion
Patch5:     0005-go1.17.13-eks-os-net-http-avoid-escapes-from-os.patch
```

Comment (markapruett) [reply projects/golang/go/1.17/rpmbuild/SPECS/golang.spec:160]: Not sure if this is a problem? Haven't seen this format before. The presubs pass here, but failed in 1.16 which has the same thing so maybe this isn't concerning.

Comment (kschumy) [reply projects/golang/go/1.17/rpmbuild/SPECS/golang.spec:160]: I don't think it is. But I can try renaming the file and see if that fixes it

**Review by markapruett — APPROVED** [review]

**Pull request merged**